drivers/gpu/drm/bridge/cdns-mhdp.h

```diff
@@ -240,6 +247,7 @@ struct cdns_mhdp_device {
 
 	struct cdns_mhdp_host host;
 	struct cdns_mhdp_sink sink;
+	struct cdns_mhdp_link_state link_state;
```

Comment:
The desc says "during link training", but I presume this is valid also after the link training?

This is a bit confusing. We already have `struct drm_dp_link link`, which sounds similar to this link_state. They seem to have overlapping fields, and in the commit you even copy the fields from the drm_dp_link to cdns_mhdp_link_state. Shouldn't we just have a single struct for this?

Reply:
Yes, but struct drm_dp_link is defined in drm framework and it does not have fields to store VS and pre-emphasis. I mainly want to store swing and pre-emphasis, which helps me to write separate function mhdp_phy_configure. Copying num_lanes and rate is just for completion and does not have actual use. Any better idea?

Reply:
Sure, but I don't know if my ideas work in practice =). But two thoughts:
- Drop drm_dp_link from the cdns_mhdp_device, and only have the cdns_mhdp_link_state. If I'm not mistaken, drm_dp_link is strictly needed only in one or two places where drm framework funcs are called. It would be easy to construct a local drm_dp_link var there, and fill in the fields from cdns_mhdp_link_state.
- Or drop the num_lanes and rate from cdns_mhdp_link_state. I think it gets confusing and error-prone when the same data is duplicated, and might not be in sync.